src/context.cc

```diff
@@ -119,6 +119,8 @@ boost::uuids::uuid Context::sim_id() {
 
 void Context::AddPrototype(std::string name, Agent* p) {
   protos_[name] = p;
+  // explicit call to support in situ prototype generation
```

Comment:
could we make this say sth like `explicit snapshot required for post-xml-init added prototypes`